src/core/index.js

```diff
@@ -275,6 +275,43 @@ util.extend( corefn, {
     return this._private.container;
   },
 
+  mount: function( container ){
+    let cy = this;
+    let _p = cy._private;
+    let options = _p.options;
+
+    if( container == null ){ return; }
+
+    if( !is.htmlElement( container ) && is.htmlElement( container[0] ) ){
+      container = container[0];
+    }
+  
+    cy.stopAnimationLoop();
+    cy.destroyRenderer();
+
+    options.container = container;
+    _p.container = container;
+
+    options.styleEnabled = true;
+    _p.styleEnabled = true;
+
+    options.renderer.name = 'canvas';
+    options.headless = false; 
+
+    cy.initRenderer( util.extend( {
+      hideEdgesOnViewport: options.hideEdgesOnViewport,
+      textureOnViewport: options.textureOnViewport,
+      wheelSensitivity: is.number( options.wheelSensitivity ) && options.wheelSensitivity > 0 ? options.wheelSensitivity : 1,
+      motionBlur: options.motionBlur === undefined ? false : options.motionBlur, // off by default
+      motionBlurOpacity: options.motionBlurOpacity === undefined ? 0.05 : options.motionBlurOpacity,
+      pixelRatio: is.number( options.pixelRatio ) && options.pixelRatio > 0 ? options.pixelRatio : undefined,
+      desktopTapThreshold: options.desktopTapThreshold === undefined ? 4 : options.desktopTapThreshold,
+      touchTapThreshold: options.touchTapThreshold === undefined ? 8 : options.touchTapThreshold
+    }, options.renderer ) );
```

Comment:
As for the later refactoring part, we probably just need to address this line (301-310).

The defaults handling could be moved into `cy.initRenderer()` so you'd just pass in `options.renderer` here and in the constructor.